R/validate_input.R

```diff
@@ -32,14 +34,19 @@ validate_input <- function(obj, argument_name, class_string, len = NULL,
     class_string <- c(class_string, "integer", "double")
   }
   
+  ## Special case grouping variable: Allow for numeric, character or factor
+  if ("groupvariable" %in% class_string) {
+    class_string <- c("factor", "character", "numeric", "integer", "double")
```

Comment:
Maybe we should have ‘class_string’ as the first entry of the new assignment? (As for numeric above) Or is „groupvariable“ so unique that it would never be combined with other class names? Probably... but in this case we could also test for equality using ‘==‘?

Reply:
Oh no we should not use == because class_string may have length > 1. don’t know, it probably does not matter because it won‘t be > length 1 if it is „groupvariable“